coriolis_openstack_utils/actions.py

```diff
@@ -753,3 +755,157 @@ def execute_operations(self):
         # done_ids = [migr["existing_migration_id"]
         #             for migr in self._completed_migrations]
         return migrations
+
+
+class SecurityGroupCreationAction(Action):
+    action_type = ACTION_TYPE_CHECK_CREATE_SECGROUP
+    NEW_SECGROUP_DESCRIPTION = (
+        "Created by the Coriolis OpenStack utilities for source security group"
+        " '%s'.")
+
+    def __init__(self, source_client, destination_client, action_payload):
+        """
+        param source_client: source client needed for fetching source
+        secgroup info
+        param destination_client: destination client needed for secgroup
+        creation
+        param action_payload: dict(): payload (params) for the action
+        must contain 'src_tenant_id' or 'src_project_id'
+                     'dest_tenant_id' / 'dest_tenant_id'
+                     'source_name'
+        """
+        self._source_client = source_client
+        self._destination_client = destination_client
+        self.payload = action_payload
+        self.subactions = []
+
+    @property
+    def secgroup_name_format(self):
+        return CONF.destination.new_secgroup_name_format
+
+    def get_source_tenant_id(self):
+        return (self.payload.get('src_tenant_id') or
```

Comment:
Also, for future reference, please don't use the `or` notation to "select" between values as it's archaic and can (and **should** imo) be removed from the spec.
In this case you could have just:
```python
return self.payload.get(
    'src_tenant_id',
    self.payload.get('src_project_id'))
```
Or, as mentioned, just write the panned out ifs:
```python
tenant_id = self.payload.get('src_project_id')
# NOTE: if both are given, 'tenant_id' will take priority
if self.payload.get('src_tenant_id') is not None:
    tenant_id = self.payload.get('src_tenant_id')
return tenant_id
```